### PR TITLE
[dns-sd] add DNS-SD server network interface binding configuration

### PR DESCRIPTION
**Pull request by simonlingoogle**
This commit allows DNS-SD server to configure the network interface to bind the UDP socket.

### CODE REVIEW

**Review by abtink — APPROVED** [review]
LGTM. 👍

**Review by wgtdkp — APPROVED** [review]
LGTM :+1:

File: src/core/config/dnssd_server.h [hunk 1]
```diff
@@ -55,6 +55,10 @@
 #define OPENTHREAD_CONFIG_DNSSD_SERVER_PORT 53
 #endif
 
+#ifndef OPENTHREAD_CONFIG_DNSSD_SERVER_BIND_NETIF
+#define OPENTHREAD_CONFIG_DNSSD_SERVER_BIND_NETIF OT_NETIF_THREAD
```
Comment (simonlingoogle) [inline src/core/config/dnssd_server.h:59]:
```suggestion
#define OPENTHREAD_CONFIG_DNSSD_SERVER_BIND_NETIF OT_NETIF_UNSPECIFIED
```

**Comment by simonlingoogle** [comment]
Thoughts on this PR? @bukepo @librasungirl

**Review by librasungirl — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Pull request merged**